source/marsGameHud.vwf.yaml

```diff
@@ -0,0 +1,468 @@
+          }
+          if ( this.frame ) {
+            context.drawImage( this.frame, position.x, position.y );
+          }
+          context.textBaseline = "top";
+          context.font = 'bold 24px Arial';
+          context.fillStyle = "rgb(255,255,255)";
+          context.textAlign = "left";
+          context.fillText( Math.round(battery), position.x + this.width + 3, position.y - 1 );
+      setUpListeners:
+  comms:
+    extends: http://vwf.example.com/hud/element.vwf
+    properties:
+      images:
+        background: "assets/images/hud/communication_bg.png"
+        frame: "assets/images/hud/communication_frame.png"
+        characterImage:
+      width: 100
+      height: 150
+      visible: true
+      enabled: true
+      alignH: left
+      alignV: bottom
+      offsetH: 10
+      offsetV: -10
+      interval: 0
+      direction: 0
+    methods:
+      draw:
+        parameters:
+          - context
+          - position
+        body: |
+          if ( this.background ) {
+            context.drawImage( this.background, position.x, position.y );
+          }
+          if ( this.characterImage ) {
+            context.save();
+            var opening = this.height * this.interval;
+            context.beginPath();
+            context.rect( position.x, position.y + ( ( this.height - opening ) / 2 ), this.width, opening );
+            context.clip();
+            context.drawImage( this.characterImage, position.x, position.y );
+            context.restore();
+            this.interval += 0.1 * this.direction;
+            if ( this.interval > 1 ) {
+              this.interval = 1;
+            } else if ( this.interval <= 0 && this.direction === -1 ) {
+              this.interval = 0;
+              this.characterImage.src = "";
+            }
+          }
+          if ( this.frame ) {
+            context.drawImage( this.frame, position.x, position.y );
+          }
+      addCharacterImage:
+      removeCharacterImage:
+  cameraSelector:
+    extends: http://vwf.example.com/hud/element.vwf
+    properties:
+      images:
+        frame: "assets/images/hud/camera_selector_frame.png"
+        activeModeIcon:
+      width: 96
+      height: 96
+      visible: true
+      enabled: true
+      alignH: right
+      alignV: top
+      offsetH: -30
+      offsetV: 30
+      isBlinking: false
+      lastBlinkTime: 0
+      blinkInterval: 0.25
+      blinkDuration: 0.25
+      activeMode:
+        set: |
+          var src;
+          switch ( value ) {
+            case "firstPerson":
+            case "thirdPerson":
+            case "topDown":
+              src = "assets/images/hud/camera_" + value + ".png";
+              break;
+            default:
+              src = "";
+              break;
+          }
+          var images = this.images;
+          images.activeModeIcon = src;
+          this.images = images;
```

Comment:
With a setter this complex, I would add a method named something like setActiveMode$, define it in the .js, and then just call it here.